Cython/Compiler/Parsing.py

```diff
@@ -2528,6 +2528,7 @@ def p_IF_statement(s: PyrexScanner, ctx):
 @cython.cfunc
 def p_statement(s: PyrexScanner, ctx, first_statement: cython.bint = 0):
     cdef_flag = ctx.cdef_flag
+    warn_deprecated_if = s.context.compiler_directives.get("warn.deprecated_if", False)
```

Comment:
Most people won't have `IF` in their code, so this should rather go into the `elif "IF"` block below.